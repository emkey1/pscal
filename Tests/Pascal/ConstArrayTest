program TestArrayConst;
uses crt; // Optional, for clrscr if desired

const
  // String array constant (like your example)
  CardValues: array[1..13] of string = (
    'A', '2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K'
  );

  // Integer array constant
  IntSequence: array[0..4] of integer = (10, -5, 100, 0, 42);

  // Simple constant for comparison
  TestStr: string = 'Simple';

var
  i: integer;
  s: string;

begin
  clrscr; // Optional clear screen

  writeln('Testing Typed Array Constants:');
  writeln('------------------------------');

  // Test simple constant (ensure it still works)
  writeln('Simple Const TestStr: ', TestStr);
  writeln;

  // Test string array constant access
  writeln('Testing CardValues (String Array):');
  writeln('CardValues[1] = ', CardValues[1], '(A)');   // Expected: A
  writeln('CardValues[10] = ', CardValues[10], '(10)'); // Expected: 10
  writeln('CardValues[13] = ', CardValues[13], '(K)'); // Expected: K

  // Assign an element to a variable
  s := CardValues[11];
  writeln('Assigned CardValues[11] to s: ', s); // Expected: J
  writeln;

  // Test integer array constant access
  writeln('Testing IntSequence (Integer Array):');
  writeln('IntSequence[0] = ', IntSequence[0], '(10)'); // Expected: 10
  writeln('IntSequence[1] = ', IntSequence[1], '(-5)'); // Expected: -5
  writeln('IntSequence[4] = ', IntSequence[4], '(42)'); // Expected: 42

  // Loop through and print integer array
  writeln('Looping through IntSequence:');
  // Note: Using hardcoded bounds as low()/high() on constants might not be implemented
  for i := 0 to 4 do
  begin
    write(IntSequence[i], ' ');
  end;
  writeln; // Newline after loop

  writeln;
  writeln('Test complete.');
end.
