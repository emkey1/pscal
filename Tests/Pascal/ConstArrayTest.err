Parser error at line 120, column 4: Exp body (found token: END)
