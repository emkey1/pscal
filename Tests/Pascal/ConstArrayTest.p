program ConstArrayTest;

const
  CardValues: array[1..3] of string = ('A','2','3');
  IntSeq: array[0..2] of integer = (10,-5,42);
  TestStr: string = 'Simple';

var
  i: integer;
  s: string;

begin
  writeln('Simple Const TestStr: ', TestStr);
  writeln('CardValues[1] = ', CardValues[1]);
  writeln('CardValues[3] = ', CardValues[3]);
  s := CardValues[2];
  writeln('Assigned CardValues[2] to s: ', s);
  for i := 0 to 2 do
    write(IntSeq[i], ' ');
  writeln;
end.
