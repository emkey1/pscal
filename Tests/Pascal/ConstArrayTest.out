Testing Typed Array Constants:
------------------------------
Simple Const TestStr: Simple

Testing CardValues (String Array):
CardValues[1] = A(A)
CardValues[10] = 10(10)
CardValues[13] = K(K)
Assigned CardValues[11] to s: J

Testing IntSequence (Integer Array):
IntSequence[0] = 10(10)
IntSequence[1] = -5(-5)
IntSequence[4] = 42(42)
Looping through IntSequence:
10 -5 100 0 42 

Test complete.
